libfreerdp/core/settings.c

```diff
@@ -991,6 +996,33 @@ BOOL freerdp_settings_copy(rdpSettings* _settings, const rdpSettings* settings)
 
 	CopyMemory(_settings->ClientTimeZone, settings->ClientTimeZone,
 			   sizeof(TIME_ZONE_INFORMATION));
+
+	_settings->RedirectionPasswordLength = settings->RedirectionPasswordLength;
+	if (settings->RedirectionPasswordLength > 0)
+	{
+		_settings->RedirectionPassword = malloc(_settings->RedirectionPasswordLength);
```

Comment:
Variable not set to NULL if nothing allocated (should already be NULL in source settings, but just to be sure)
(same below)

Reply:
so if settings->RedirectionPasswordLength == 0 -> _settings->RedirectionPassword needs to be set to NULL. right?

Reply:
@akallabeth

Reply:
by the way, i think this is okay, because at the begining of the function all pointers are copied (the deref copy *_settings = *settings), so the length will be 0 and the pointer will remain NULL